.github/actions/gatsby-site-showcase-validator/index.js

```diff
@@ -47,34 +48,47 @@ async function run() {
     }
 
     // Pass html into a parser
-    let $ = cheerio.load(siteHtml)
+    const $ = cheerio.load(siteHtml)
 
     // Most Gatsby sites have an id of "___gatsby"
-    let gatsbyContainer = $("#___gatsby")
+    const gatsbyContainer = $("#___gatsby")
 
-    if (gatsbyContainer.length !== 0) {
-      // The site is a gatsby site don't do anything
-      sitesVisited++
-    } else {
-      // The site is not a gatsby site, print out some info
+    // The site is not a gatsby site, print out some info
+    if (!gatsbyContainer.length) {
       console.log(
         `${chalk.yellow(`[Notice]`)}: ${
           site.title
         } (${siteUrl}) is not a Gatsby site`
       )
-      sitesVisited++
       nonGatsbySiteCount++
     }
+
+    // Check if provided repository is public
+    if (sourceUrl) {
+      const status = await fetch(sourceUrl).then(({ status }) => status)
+
+      if (status !== 200) {
+        console.log(
+          `${chalk.yellow(`[Notice]`)}: ${
+            site.title
+          } (${siteUrl}) provided a 'source_url', but it's repository is private (${sourceUrl})`
+        )
+        nonPublicRepoCount++
+      }
+    }
+
+    sitesVisited++
   }
 
   console.log(
     chalk.green(
-      `We visited ${sitesVisited}/${totalSitesCount} sites. Out of them, ${nonGatsbySiteCount} sites were not a Gatsby site and ${erroredOut} errored out when visiting it.`
+      `We visited ${sitesVisited}/${totalSitesCount} sites. Out of them, ${nonGatsbySiteCount} sites were not a Gatsby site, ${nonPublicRepoCount} provided private repositories, and ${erroredOut} errored out when visiting it.`
     )
   )
 
-  // If there are any non Gatsby sites, fail (non-zero exit code)
-  process.exit(nonGatsbySiteCount > 0 ? 1 : 0)
+  // If there are any non Gatsby sites or their `source_url` is private, fail (non-zero exit code)
+  const exitCode = nonGatsbySiteCount > 0 || nonPublicRepoCount > 0 ? 1 : 0
```

Comment:
i don't know exactly the execution order but this is clearer: 
```suggestion
  const exitCode = (nonGatsbySiteCount > 0 || nonPublicRepoCount > 0) ? 1 : 0
```

Reply:
execution order is from left to right and there's only one operator so this seems redundant to me

Reply:
I personally would go for more easy debug for the next one then for redundant chars.

Reply:
But that's my personal taste